staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/maxitems/doc.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +k8s:validation-gen=TypeMeta
+// +k8s:validation-gen-scheme-registry=k8s.io/code-generator/cmd/validation-gen/testscheme.Scheme
+
+// This is a test package.
+package maxitems
+
+import "k8s.io/code-generator/cmd/validation-gen/testscheme"
+
+var localSchemeBuilder = testscheme.New()
+
+// Maxitems
+type M struct {
+	TypeMeta int
+
+	// +k8s:validation:maxItems=3
+	M1 []T `json:"m1"`
```

Comment:
Super minor, but in most of the test this would be "LT" (list of T) and the below would be `M2 M2`

Reply:
Good question - I think we always check the type-attached rules before field-attached ones, so we would validate M1 against 2, then 3 and M2 against 2, then 1.  I think it would be fine to issue two errors here because this is not a "real" case.

Reply:
In the current generation code, it would check the field-attached ones first then to type attached ones. I think it make more sense as the field could have more restricted limitation than the types.

Reply:
Yes, I was wrong, thanks.  Also I missed that this was marked ShortCircuit